com.incquerylabs.v4md/src/main/com/incquerylabs/v4md/ViatraQueryAdapter.java

```diff
@@ -124,11 +126,13 @@ public void executeActionOnEngine(Consumer<AdvancedViatraQueryEngine> action) {
 			isInitialized = engine.map(e -> {
 					boolean thereWasException = false;
 					try {
-						e.getBaseIndex().coalesceTraversals(() -> {
-							initializationActions.forEach(action -> action.accept(e));
-							return null;
-						});
-						initializationActions.clear();
+						if(!initializationActions.isEmpty()) {
```

Comment:
It would have been nice for this update to be mentioned in the commit comment as well. It makes sense to implement it, but this seems unrelated to the fix itself.

I know this is not really possible to fix in Github, but in Gerrit I would have asked to do this in a separate change/commit.